lucene/core/src/java/org/apache/lucene/index/IndexingChain.java

```diff
@@ -399,7 +401,8 @@ private void writeDocValues(SegmentWriteState state, Sorter.DocMap sortMap) thro
             }
             perField.docValuesWriter.flush(state, sortMap, dvConsumer);
             perField.docValuesWriter = null;
-          } else if (perField.fieldInfo.getDocValuesType() != DocValuesType.NONE) {
+          } else if (perField.fieldInfo != null
```

Comment:
shouldn't fieldInfo always be non null when reaching here?

Reply:
Not, always.
We first create PerField: `PerField pf = getOrAddPerField(field.name(), fieldType);` and then initialize its FieldInfo `initializeFieldInfo(pf)`. Initialization of FieldInfo may fail, if there is already a global field with the same name but a different schema. So we may end up with `PerField` with a null `fieldInfo`.